ca/certificate-authority.go

```diff
@@ -381,35 +463,45 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 		// AUDIT[ Error Conditions ] 9cc4d537-8534-4970-8665-4b382abe82f3
 		ca.log.Audit(fmt.Sprintf("Serial increment failed, rolling back: err=[%v]", err))
 		tx.Rollback()
-		return emptyCert, err
+		return
 	}
 	serialHex := fmt.Sprintf("%02X%014X", ca.Prefix, serialDec)
 
 	// Send the cert off for signing
+	csrPEM := string(pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE REQUEST",
+		Bytes: sr.csr.Raw,
+	}))
 	req := signer.SignRequest{
 		Request: csrPEM,
 		Profile: ca.profile,
-		Hosts:   hostNames,
+		Hosts:   sr.hostNames,
 		Subject: &signer.Subject{
-			CN: commonName,
+			CN: sr.commonName,
 		},
 		SerialSeq: serialHex,
 	}
 
+	// Mark the cert valid for now.  We will finalize to this cert if
+	// things go wrong
+	sr.cert.Status = core.StatusInvalid
```

Comment:
Using `sr.cert` all through here instead of a dedicated object seems kind of clunky, but probably tolerable.